src/JSON.ts

```diff
@@ -498,6 +498,11 @@ export abstract class Value {
     }
 
     public abstract accept(visitor: Visitor): void;
+
+    /**
+     * A user-friendly string that represents this value, suitable for display in a label, etc.
+     */
+    public abstract toFriendlyString(): string;
```

Comment:
If it is being used solely for labels, I'd name it `toLabel`. What friendly for labels may not be friendly for filename titles.

Reply:
I'm not sure it will be solely used for labels, thus etc.  I don't want to have something specific to just labels in the base classes, this makes it a bit more generic, comment specifies intent to help decide future direction when additional use cases are clear.